ui/lib/core/addon/components/filter-input.ts

```diff
@@ -38,11 +25,10 @@ export default class FilterInputComponent extends Component<Args> {
 
   @action
   onInput(event: HTMLElementEvent<HTMLInputElement>) {
-    const callback = () => {
-      this.args.onInput(event.target.value);
-    };
-    const wait = this.args.wait || 200;
+    const wait = this.args.wait || 500;
     // ts complains when trying to pass object of optional args to callback as 3rd arg to debounce
-    debounce(this, callback, wait);
+    // eslint-disable-next-line
+    // @ts-ignore
+    debounce(this, this.args.onInput, event.target.value, wait);
   }
```

Comment:
It seems like the callback was firing without waiting for the debounce which is why I refactored this. The ts definition for `debounce` must be wrong because it complains about the optional 3rd arg. It wants it to be the wait time. In any case this aligns with the examples in the docs and it works so I opted to ignore the warnings.